packages/vite/src/shared/hmr.ts

```diff
@@ -219,12 +219,14 @@ export class HMRClient {
         if (disposer) return disposer(this.dataMap.get(path))
       }),
     )
-    paths.forEach((path) => {
-      const fn = this.pruneMap.get(path)
-      if (fn) {
-        fn(this.dataMap.get(path))
-      }
-    })
+    await Promise.all(
+      paths.map((path) => {
+        const fn = this.pruneMap.get(path)
+        if (fn) {
+          fn(this.dataMap.get(path))
```

Comment:
```suggestion
          return fn(this.dataMap.get(path))
```

Shouldn't this need to return the promise?